src/configurator.ts

```diff
@@ -119,6 +120,57 @@ export class Configurator {
 
     core.addPath(toolPath);
   }
+
+  validate() {
+    if (this.name === "") {
+      throw new Error(
+        `the name used to configure the tool in path is required.`
+      );
+    }
+
+    if (!this.fromGitHubReleases && this.url === "") {
+      throw new Error(`url is required if trying to fetch tool directly.`);
+    }
+
+    if (!this.fromGitHubReleases && !matchesUrlRegex(this.url)) {
+      throw new Error(`url supplied as input is not a valid URL.`);
+    }
+
+    if (this.fromGitHubReleases && !matchesUrlRegex(this.urlTemplate)) {
+      throw new Error(`urlTemplate supplied as input is not a valid URL.`);
```

Comment:
Dumb question: have you tested this works with urls with `{{ }}` in them?

Reply:
Yeah, tested that manually, plus there is one test case for it (there should probably be more than just one..) - https://github.com/radu-matei/configurator/blob/input-validation/test/action.ts#L143-L163